include/seqan3/range/container/aligned_sequence_adaptor_constant_access.hpp

```diff
@@ -0,0 +1,688 @@
+#include <seqan3/range/container/concept.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+
+#include "sdsl/bit_vectors.hpp"
+#include "sdsl/rank_support_v5.hpp"
+#include "sdsl/select_support_mcl.hpp"
+#include "sdsl/util.hpp"
+
+namespace seqan3 {
+
+/*!\brief Container for an aligned sequence with random access at constant time.
+ * \tparam inner_type {The container type with which aligned sequence structure
+ * can be constructed, e.g. std::vector<gapped<dna4>>. The inner type of the
+ * container must fulfill the seqan3::alphabet_concept.}
+ * \implements partially the seqan3::random_access_sequence_concept
+ * \remark {This class has to be parameterized with the union type
+ * gapped<alphabet_type> packed into a container supporting the random access
+ * operators [] and at(). This way an aligned sequence can be initialized or
+ * assigned to any sequence of alphabet and/or gap symbols.}
+ *
+ * The gap information of an aligned sequence is internally stored in a bit_vector
+ * set to one whenever an aligned position corresponds to a gap. The ungapped
+ * sequence making up the 0s in the bit vector is stored 'compressed' in a
+ * std::vector.
+ * We use rank and a select functions of the sdsl to map from the gapped
+ * sequence space to the ungapped space and vice versa.
+ * \par Example
+ * '---AT--ATC-GT' is represensed by the bit vector '1110011000100' and the
+ * compressed sequence 'ATATCGT'.
+ */
+
+//!\brief Random access operator concept.
+ template <typename type>
+ concept bool random_access_concept = requires (type val)
+ {
+     // member types
+     typename type::reference;
+     // access container
+     { val[0]    } -> typename type::reference;
+     { val.at(0) } -> typename type::reference;
+ };
+
+template <typename inner_type>
+//!\cond
+    //requires random_access_sequence_concept<inner_t> && alphabet_concept<ranges::v3::value_type_t<inner_type>>
+    requires random_access_concept<inner_type> &&
+    alphabet_concept<ranges::v3::value_type_t<inner_type>>
+//!\endcond
+struct aligned_sequence_adaptor_constant_access
+{
+
+private:
+     //!\privatesection
+    using aligned_sequence_t = aligned_sequence_adaptor_constant_access;
+    //using alphabet_t = typename ranges::v3::value_type_t<inner_type>::alphabet_type;
+
+// re-initialize rank support structure of bit_vector
+void update_support_structures()
+{
+    rs = sdsl::rank_support_v5<1, 1>(&gap_vector);
+    ss = sdsl::select_support_mcl<0, 1>(&gap_vector);
+}
+
+public:
+    //!\publicsection
+    /*!\name Member types
+    * \{
+    */
+    //!\brief Value type of container elements.
+    //!\hideinitializer
+    using value_type = typename ranges::v3::value_type_t<inner_type>;
+
+    //!\brief Use reference type defined by container.
+    //!\hideinitializer
+    using reference = value_type;
+
+    //!\brief Use const reference type provided by container.
+    //!\hideinitializer
+    using const_reference = const reference;
```

Comment:
this should just be value_type, too! No need to have const, since new values are created anyway!